In [ ]:
#模型下载
from modelscope import snapshot_download
model_dir = snapshot_download('BAAI/bge-m3', cache_dir='D:\\AInewModels')


In [4]:
!pip install tf-keras

Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple
     ---------------------------------------- 0.0/1.7 MB ? eta -:--:--
     ---------------------------------------- 1.7/1.7 MB 15.6 MB/s eta 0:00:00


In [ ]:
!pip install FlagEmbedding


Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple


In [5]:
from FlagEmbedding import BGEM3FlagModel

model = BGEM3FlagModel('D:\\AInewModels\\BAAI\\bge-m3',  
                       use_fp16=True) # Setting use_fp16 to True speeds up computation with a slight performance degradation

sentences_1 = ["What is BGE M3?", "Defination of BM25"]
sentences_2 = ["BGE M3 is an embedding model supporting dense retrieval, lexical matching and multi-vector interaction.", 
               "BM25 is a bag-of-words retrieval function that ranks a set of documents based on the query terms appearing in each document"]

embeddings_1 = model.encode(sentences_1, 
                            batch_size=12, 
                            #max_length=8192, # If you don't need such a long length, you can set a smaller value to speed up the encoding process.
                            max_length=1024,
                            )['dense_vecs']
embeddings_2 = model.encode(sentences_2)['dense_vecs']
similarity = embeddings_1 @ embeddings_2.T
print(similarity)

You're using a XLMRobertaTokenizerFast tokenizer. Please note that with a fast tokenizer, using the `__call__` method is faster than using a method to encode the text followed by a call to the `pad` method to get a padded encoding.


[[0.6259036  0.34749582]
 [0.34986773 0.6782464 ]]


- "What is BGE M3?" 与"BGE M3 is an embedding model..." 的相似度为0.6265（较高）
- "What is BGE M3?"与"- BM25 is a bag-of-words retrieval function..." 的相似度为0.3477（较低）
- "Defination of BM25" 与"BGE M3 is an embedding model..." 的相似度为0.3499（较低）
- "Defination of BM25" 与"BM25 is a bag-of-words retrieval function..." 的相似度为0.678（较高）